sklearn/cluster/tests/test_hdbscan.py

```diff
@@ -22,12 +22,27 @@
 from sklearn.utils import shuffle
 from sklearn.utils._testing import assert_allclose, assert_array_equal
 
+# TODO(1.6): Remove
+filterwarnings_kdtree = pytest.mark.filterwarnings(
+    "ignore:`algorithm='kdtree'`has been deprecated in 1.4 and will be renamed"
+    " to'kd_tree'`in 1.6. To keep the past behaviour, set `algorithm=kd_tree`."
+)
+# TODO(1.6): Remove
+filterwarnings_balltree = pytest.mark.filterwarnings(
+    "ignore:`algorithm='balltree'`has been deprecated in 1.4 and will be renamed"
+    " to'ball_tree'`in 1.6. To keep the past behaviour, set `algorithm=ball_tree`."
+)
+
```

Comment:
Rather than filterwarnings everywhere, we can go ahead and use the new algorithm names. We need only test that the deprecation works as expected in one place and move on to future behavior everywhere else.